src/panel.tsx

```diff
@@ -14,7 +14,15 @@ const Iframe = styled.iframe({
 });
 
 function parseShareURL(url: string): string | void {
-  return url.split(/share.(?:go)?abstract.com\//)[1];
+  if (url.match(/share.(?:go)?abstract.com\//)) {
+    const parsedUrl = new URL(url);
+    const pathSegments = parsedUrl.pathname.split('/');
+    return pathSegments[pathSegments.length - 1];
+  }
+
+  throw new Error(
+    `The provided url (${url}) is not valid. The url must come from "https://share.abstract.com/" or "https://share.goabstract.com/".`,
+  );
```

Comment:
No need to throw here. Checkout the return type, it is okay to return nothing here

If anything we can let the caller throw if it receives nothing from this function